### PR TITLE
Refreshing the docs

### PR DESCRIPTION
**Pull request by jcrossley3**
Our docs were getting a little dated. Here's my attempt at re-organizing them to be more useful to new users.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+the `controller` container in the `broker` deployment spec, and the
+value of the `DISPATCHER_IMAGE` env var.
+
+```
+spec:
+  registry:
+    override:
+      autoscaler: docker.io/my-org/autoscaler:v0.13.0
+      broker/controller: docker.io/my-org/broker-controller:v0.13.0
+      DISPATCHER_IMAGE: docker.io/my-org/dispatcher:v0.13.0
```
Comment (aliok) [inline docs/configuration.md:104]: [not a comment on the doc]

Oh boy. This is confusing :(

Not really related to how the documentation is. Documentation is good.
The image override mechanism is confusing. 3 different patterns.

We should've accepted @pmorie 's suggestion of creating typed API for these

**Comment by aliok** [comment]
cc @abrennan89 maybe you're interested

File: docs/configuration.md (hunk 1 above)
Comment (jcrossley3) [reply docs/configuration.md:104]: I agree, fwiw, but I also think the `registry` field is an advanced use case.

**Pull request merged**